tasks/mainternace.ts

```diff
@@ -433,9 +433,39 @@ export function load() {
     console.log(`EUR/USD: ${price} last updated: ${new Date(lastUpdateTimestamp * 1000)}`);
   }
 
-  async function show_efBDXCov(stable: BDStable) {
-    const efBdxCov = await stable.get_effective_bdx_coverage_ratio();
-    console.log(`${await stable.symbol()} efBDXCov: ${d12_ToNumber(efBdxCov.mul(100))}%`);
+  async function show_efBDXCov(hre: HardhatRuntimeEnvironment, stable: BDStable) {
+    const [cr_d12, eCr, totalSupply, symbol, bdxPrice_d12, bdx, unclaimedPoolsBDX] = await Promise.all([
+      stable.global_collateral_ratio_d12(),
+      stable.effective_global_collateral_ratio_d12(),
+      stable.totalSupply(),
+      stable.symbol(),
+      stable.BDX_price_d12(),
+      getBdx(hre),
+      stable.unclaimedPoolsBDX()
+    ]);
+
+    const usedCr_d12 = cr_d12.gt(eCr) ? eCr : cr_d12;
+    const COLLATERAL_RATIO_MAX = BigNumber.from(1e12);
+    const COLLATERAL_RATIO_PRECISION = BigNumber.from(1e12);
+    const PRICE_PRECISION = BigNumber.from(1e12);
+
+    const expectedBdxValue_d18 = COLLATERAL_RATIO_MAX.sub(usedCr_d12).mul(totalSupply).div(COLLATERAL_RATIO_PRECISION);
+
+    // If we don't need BDX, the coverage ratio is 100%
+    if (expectedBdxValue_d18.eq(0)) {
+      console.log(`${symbol} efBDXCov: 100%`);
+    } else {
+      // BDX hit rock bottom
+      if (bdxPrice_d12.eq(0)) {
+        console.log(`${symbol} efBDXCov: 0%`);
+      } else {
+        const expectedBdx_d18 = expectedBdxValue_d18.mul(PRICE_PRECISION).div(bdxPrice_d12);
+
+        const bdxSupply_d18 = (await bdx.balanceOf(formatAddress(hre, stable.address))).sub(unclaimedPoolsBDX);
+        const effectiveBdxCR_d12 = PRICE_PRECISION.mul(bdxSupply_d18).div(expectedBdx_d18);
+        console.log(`${symbol} efBDXCov: ${d12_ToNumber(effectiveBdxCR_d12.mul(100))}%`);
```

Comment:
let's print also the bdx balance

Reply:
BDEU efBDXCov: 116.9211585353% | BDX supply: 618,700.32740
BDUS efBDXCov: 124.3983050576% | BDX supply: 1,179,454.04066
bXAU efBDXCov: 237.292751111% | BDX supply: 267,870.82129
bGBP efBDXCov: 156.2015943985% | BDX supply: 78,267.93937